src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -218,13 +218,18 @@ def uc_grant_sql(self, object_type: str | None = None, object_key: str | None =
 class GrantsCrawler(CrawlerBase[Grant]):
     """Crawler that captures access controls that relate to data and other securable objects."""
 
-    def __init__(self, tables_crawler: TablesCrawler, udf: UdfsCrawler, include_databases: list[str] | None = None):
-        assert tables_crawler._sql_backend == udf._sql_backend
-        assert tables_crawler._catalog == udf._catalog
-        assert tables_crawler._schema == udf._schema
-        super().__init__(tables_crawler._sql_backend, tables_crawler._catalog, tables_crawler._schema, "grants", Grant)
+    def __init__(
```

Comment:
I decoupled the grantscrawler from the tables crawler for two reasons:
1. IMO it is confusing because AFAIK this is the only crawler which does it. It also bears the question: why not tightly couple it with the UDFs crawler? 
2. The `GrantsCrawler` is expected to run against a different `SqlBackend` in the assessment workflow, namely the TACL cluster.